instance/tests/management/test_kill_zombies.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Tests for the 'kill_zombies' management command.
+"""
+# Imports #####################################################################
+
+from unittest.mock import patch, call
+from django.core import mail
+from django.core.management import CommandError
+from django.test import TestCase, override_settings
+from instance.tasks import KillZombiesRunner
+
+# Tests #######################################################################
+
+# pylint: disable=bad-continuation
+
+
+@override_settings(
+    KILL_ZOMBIES_ENABLED=True,
+    ADMINS=(
+        ("admin1", "admin1@localhost"),
+        ("admin2", "admin2@localhost"),
+    ),
+    OPENSTACK_REGION="some-region"
+)
+class KillZombiesPeriodicallyTestCase(TestCase):
+    """
+    Test cases for the `kill_zombies_task` periodic task.
+    """
+    @override_settings(KILL_ZOMBIES_ENABLED=False)
+    @patch("instance.tasks.call_command")
+    def test_toggle_task_false(self, mock_command):
+        """
+        Tests that the task can't be run if disabled
+        through env variables
+        """
+        with self.assertRaises(ImportError):
+            from instance.tasks import kill_zombies_task
+            kill_zombies_task()
+        self.assertFalse(mock_command.called)
+
+    @patch("instance.tasks.call_command")
+    def test_set_env_variables(self, mock_command):
+        """
+        Tests that the task passes env variables to
+        the kill_zombies command call
+        """
+        test_runner = KillZombiesRunner()
+        test_runner.run()
+        first_call = call.mock_command("kill_zombies", region="some-region", dry_run=True)
+        self.assertEqual(mock_command.mock_calls[0], first_call)
+        mock_command.assert_called_with("kill_zombies", region="some-region")
```

Comment:
@spokerman12, does this check the last call of the mock command? This is relevant because the previous statement checks the first call and we shouldn't be checking the same again.

Reply:
If this doesn't test the expected thing, there is a similar assertion below that may have to be modified as well.

Reply:
Yes, `assert_called_with` only checks the last call (so last call is not a dry_run), that's why I define a `first_call` then use `assertEqual` to compare with the `mock_calls[0]`, which is the first that happened.